dotCMS/src/main/java/com/dotcms/rest/api/v1/storage/FileStorageResource.java

```diff
@@ -0,0 +1,486 @@
+                            zoutput.putNextEntry(new ZipEntry(path));
+
+                            while ((bytesRead = fileInput.read(buffer)) > 0) {
+
+                                zoutput.write(buffer, 0, bytesRead);
+                            }
+
+                            zoutput.closeEntry();
+                        } catch (Exception e) {
+
+                            Logger.warn(this.getClass(), e.getMessage(), e);
+                            throw new DotRuntimeException(e.getMessage());
+                        }
+                    }
+                },
+                "application/zip")
+                .header("content-disposition", "attachment; filename=" + fileName + ".zip")
+                .build();
+    }
+
+
+    @POST
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    public final Response addFileToStorage(@Context final HttpServletRequest request,
+                                           @Context final HttpServletResponse response,
+                                           final FormDataMultiPart multipart) throws IOException, JSONException {
+
+        final Tuple2<Map<String,Object>, List<File>> bodyAndBinaries =
+                this.multiPartUtils.getBodyMapAndBinariesFromMultipart(multipart);
+        final List<File> files            = bodyAndBinaries._2();
+
+        final ImmutableMap.Builder<File, Object> bodyResultBuilder = new ImmutableMap.Builder<>();
+        if (!UtilMethods.isSet(files)) {
+
+            throw new BadRequestException("Must send files");
+        }
+
+        for (final File file : files) {
+
+            final String storageType = Config.getStringProperty("DEFAULT_STORAGE_TYPE", StorageType.FILE_SYSTEM.name());
+            bodyResultBuilder.put(file, this.fileStorageAPI.getStorageProvider().getStorage(storageType)
+                    .pushFile("files", File.separator + file.getName(), file, Collections.emptyMap()));
+        }
+
+        return Response.ok(new ResponseEntityView(bodyResultBuilder.build())).build();
+    }
+
+
+    @PUT
+    @Path("/metadata/basic")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    public final Response generateMetadata(@Context final HttpServletRequest request,
+                                      @Context final HttpServletResponse response,
+                                               final FormDataMultiPart multipart) throws IOException, JSONException {
+
+        final Tuple2<Map<String,Object>, List<File>> bodyAndBinaries =
+                this.multiPartUtils.getBodyMapAndBinariesFromMultipart(multipart);
+        final List<File> files            = bodyAndBinaries._2();
+        final Map<String, Object> bodyMap = bodyAndBinaries._1();
+
+        final ImmutableMap.Builder<String, Object> bodyResultBuilder = new ImmutableMap.Builder<>();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5547873)